server/api/src/classifier.py

```diff
@@ -1,24 +1,149 @@
 """
-We'll add something interesting a bit later now let's mock the response
+The requestor's agent controlling and interacting with the classifier vm
 """
+import asyncio
+from datetime import datetime, timedelta
+import aiofiles
+import os
+from itertools import count
 
-from typing import Dict
-
-
-def classify_text(text: str) -> Dict:
-    return {
-        "results": [
-            {
-                "classifier_id": "distilbart-cnn-12-6",
-                "summary_text": " ".join(text.lower().split()[:10])
-            },
-            {
-                "classifier_id": "bert-base-uncased-emotion",
-                "summary_list": [
-                    { "label": "love", "score": 0.005923508666455746, },
-                    { "label": "joy", "score": 0.38369229435920715, },
-                    { "label": "surprise", "score": 0.6015767455101013, },
-                ]
-            },
-        ],
+from yapapi import (
+    Executor,
+    Task,
+    __version__ as yapapi_version,
+    WorkContext,
+)
+
+from yapapi.log import enable_default_logger, log_summary, log_event_repr  # noqa
+from yapapi.package import vm
+
+
+async def service_start(yagna_app):
+    # Set the Yagna's APP_KEY
+    assert yagna_app and yagna_app["appkey"], "Classificator cannot be started, Yagna's APP_KEY is missing."
+    os.environ["YAGNA_APPKEY"] = yagna_app["appkey"]
+
+    timeout = timedelta(minutes=30)
+    package = await vm.repo(
+        image_hash=os.getenv("PROVIDER_IMAGE_HASH", "c6b743459d3428fb860582e556ceba1c76dbc8a1d599a55dcf73e437"),
+        min_mem_gib=1.5,
+        min_storage_gib=3.0,
+    )
+
+    async def handle_requests(ctx: WorkContext, tasks):
+        """
+        Initializes classifiers and yields texts as tasks
+        """
+        text_queue = yagna_app["tasks"]
+
+        task = await tasks.__anext__()
+
+        try:
+            ctx.run("/bin/sh", "-c", "nohup python classifier.py run &")
+            print("Activity initialized.")
+
+            for i in count(1):
+                test_filename = f"sample_{i:08}.txt"
+
+                text, fut = await text_queue.get()
+                assert text, "Empty input not expected"
+                print(f"Received text from queue: {text[:24]}...")
+
+                input_file_path = f"./{test_filename}.in"
+                async with aiofiles.open(input_file_path, mode="w") as f:
+                    await f.write(text)
```

Comment:
That's how you should handle any I/O if you're writing asynchronous code. Reading/writing files works orders of magnitude slower than executing code. If you used blocking I/O operations you would block the whole thread. If your code is asynchronous you need a good reason to use synchronous I/O, async is the default option.

Reply:
I disagree with 
> We don't want to pass the execution to any other part.

another web request can be served while it (a)waits for a file here.

> We don't even really run any async code here (the app is awaiting either way).

We do run it in the async manner, app can answer to `GET /api` requests ... I'm not sure regarding to another text for classifier ... maybe we should block it just in case 😸